CommandLine/libEGM/gmk.cpp

```diff
@@ -49,380 +50,6 @@ namespace gmk_internal {
-    return;
-  }
-  unsigned pixeloffset = bmp[10] + 256 * bmp[11]; //where the pixel data starts
-  //read width and height from BMP header
-  size_t w = bmp[18] + bmp[19] * 256;
-  size_t h = bmp[22] + bmp[23] * 256;
-  //read number of channels from BMP header
-  if (bmp[28] != 24 && bmp[28] != 32) {
-    errStream << "Image from the GMK file was " << bmp[28] << " bit while this reader "
-        << "only supports 24 and 32 bit BMP images" << std::endl;
-    return;
-  }
-  unsigned numChannels = bmp[28] / 8;
-
-  //The amount of scanline bytes is width of image times channels, with extra bytes added if needed
-  //to make it a multiple of 4 bytes.
-  unsigned scanlineBytes = w * numChannels;
-  if (scanlineBytes % 4 != 0) scanlineBytes = (scanlineBytes / 4) * 4 + 4;
-
-  unsigned dataSize = scanlineBytes * h;
-  if (length < dataSize + pixeloffset) {
-    errStream << "Image from the GMK file had a length '" << length << "' smaller than the estimated "
-        << "size '" << (dataSize + pixeloffset) << "' based on the BMP header dimensions" << std::endl;
-    return;
-  }
-
-  std::vector<unsigned char> rgba(w * h * 4);
-
-  // get the bottom left pixel for transparency
-  // NOTE: bmp is obviously upside down, so it's just the first pixel
-  unsigned char t_pixel_r = bmp[pixeloffset+0],
-                t_pixel_g = bmp[pixeloffset+1],
-                t_pixel_b = bmp[pixeloffset+2];
-
-  /*
-  There are 3 differences between BMP and the raw image buffer for libpng:
-  -it's upside down
-  -it's in BGR instead of RGB format (or BRGA instead of RGBA)
-  -each scanline has padding bytes to make it a multiple of 4 if needed
-  The 2D for loop below does all these 3 conversions at once.
-  */
-  for (unsigned y = 0; y < h; y++) {
-    for (unsigned x = 0; x < w; x++) {
-      //pixel start byte position in the BMP
-      unsigned bmpos = pixeloffset + (h - y - 1) * scanlineBytes + numChannels * x;
-      //pixel start byte position in the new raw image
-      unsigned newpos = 4 * y * w + 4 * x;
-      if (numChannels == 3) {
-        rgba[newpos + 0] = bmp[bmpos + 2]; //R<-B
-        rgba[newpos + 1] = bmp[bmpos + 1]; //G<-G
-        rgba[newpos + 2] = bmp[bmpos + 0]; //B<-R
-        rgba[newpos + 3] = 255;            //A<-A
-      } else {
-        rgba[newpos + 0] = bmp[bmpos + 3]; //R<-A
-        rgba[newpos + 1] = bmp[bmpos + 2]; //G<-B
-        rgba[newpos + 2] = bmp[bmpos + 1]; //B<-G
-        rgba[newpos + 3] = bmp[bmpos + 0]; //A<-R
-      }
-
-      if (transparent &&
-        bmp[bmpos + 0] == t_pixel_r &&
-        bmp[bmpos + 1] == t_pixel_g &&
-        bmp[bmpos + 2] == t_pixel_b) {
-        rgba[newpos + 3] = 0; //A<-A
-      }
-    }
-  }
-
-  CreateDirectoryRecursive(write_fname);
-  libpng_encode32_file(rgba.data(), w, h, write_fname.u8string().c_str());
-}
-
-void writeTempBGRAFile(const std::filesystem::path& write_fname, std::unique_ptr<char[]> bytes, size_t width, size_t height) {
```

Comment:
Where did this logic go? Was it unused...? It does look a little hacky.

Reply:
I copied it to zlib_util.cpp where its used, just didn't put it in Decoder class, as it wasn't there previously as well. Its static right now.